src/main/java/com/specialwarriors/conal/common/auth/session/SessionManager.java

```diff
@@ -0,0 +1,29 @@
+package com.specialwarriors.conal.common.auth.session;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpSession;
+import java.util.Optional;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SessionManager {
+
+    public void createSession(HttpServletRequest request, Long userId) {
+        HttpSession session = request.getSession();
+        session.setAttribute("userId", userId);
+    }
+
+    public Long getUserId(HttpServletRequest request) {
+        HttpSession session = request.getSession(false);
+
+        return (Long) Optional.ofNullable(session)
+                .map(s -> s.getAttribute("userId")).orElse(null);
```

Comment:
```suggestion
        return (Long) Optional.ofNullable(session)
                .map(s -> s.getAttribute("userId"))
                .map(Long.class::cast)
                .orElse(null);
```
캐스팅도 Optional의 메서드 체이닝을 이용하는 것은 어떨까요?